src/components/organisms/AssetActions/Edit/EditComputeDataset.tsx

```diff
@@ -0,0 +1,249 @@
+import FormEditComputeDataset from './FormEditComputeDataset'
+import styles from './index.module.css'
+import {
+  Logger,
+  DDO,
+  ServiceComputePrivacy,
+  publisherTrustedAlgorithm as PublisherTrustedAlgorithm
+} from '@oceanprotocol/lib'
+import MetadataFeedback from '../../../molecules/MetadataFeedback'
+import { graphql, useStaticQuery } from 'gatsby'
+import axios from 'axios'
+import { queryMetadata, getAssetsNames } from '../../../../utils/aquarius'
+import Loader from '../../../atoms/Loader'
+import web3 from 'web3'
+
+const contentQuery = graphql`
+  query EditComputeDataQuery {
+    content: allFile(
+      filter: { relativePath: { eq: "pages/editComputeDataset.json" } }
+    ) {
+      edges {
+        node {
+          childPagesJson {
+            description
+            form {
+              success
+              successAction
+              error
+              data {
+                name
+                placeholder
+                label
+                help
+                type
+                required
+                sortOptions
+                options
+                multiple
+                rows
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`
+
+export default function EditComputeDataset({
+  setShowEdit
+}: {
+  setShowEdit: (show: boolean) => void
+}): ReactElement {
+  const data = useStaticQuery(contentQuery)
+  const content = data.content.edges[0].node.childPagesJson
+
+  const { debug } = useUserPreferences()
+  const { ocean, accountId, config } = useOcean()
+  const { metadata, ddo, refreshDdo } = useAsset()
+  const [success, setSuccess] = useState<string>()
+  const [error, setError] = useState<string>()
+  const [algorithms, setAlgorithms] = useState<AlgorithmOption[]>()
+  const query = {
+    page: 1,
+    query: {
+      nativeSearch: 1,
+      query_string: {
+        query: `(service.attributes.main.type:algorithm) -isInPurgatory:true`
+      }
+    },
+    sort: { created: -1 }
+  }
+  const source = axios.CancelToken.source()
+
+  const hasFeedback = error || success
+
+  interface AlgorithmOption {
+    did: string
+    name: string
+  }
+
+  async function getAlgorithms() {
+    const didList: string[] = []
+    const result = await queryMetadata(
+      query as any,
+      config.metadataCacheUri,
+      source.token
+    )
+    await result.results.forEach((ddo: DDO) => {
+      const did: string = web3.utils
+        .toChecksumAddress(ddo.dataToken)
+        .replace('0x', 'did:op:')
+      didList.push(did)
+    })
+    const ddoNames = await getAssetsNames(
+      didList,
+      config.metadataCacheUri,
+      source.token
+    )
+    const algorithmList: AlgorithmOption[] = []
+    didList.forEach((did: string) => {
+      algorithmList.push({
+        did: did,
+        name: ddoNames[did]
+      })
+    })
+    setAlgorithms(algorithmList)
+  }
+
+  useEffect(() => {
+    getAlgorithms()
+  }, [])
+
+  async function createTrustedAlgorithmList(
```

Comment:
same for this method, this does not seem to rely on any changes within this component so we can just move it out